src/main/java/com/truelayer/java/payments/entities/AuthorizationFlowResponse.java

```diff
@@ -53,6 +51,18 @@ public AuthorizationFlowAuthorizationFailed asAuthorizationFailed() {
         return (AuthorizationFlowAuthorizationFailed) this;
     }
 
+    // @deprecated
+    // Only `Authorizating` responses have `AuthorizationFlow`.
+    // Use `asAuthorizing().getAuthorizationFlow()` instead.
+    @Deprecated
+    @JsonIgnore
+    public AuthorizationFlow getAuthorizationFlow() {
```

Comment:
I'd be ok deprecating this, but this `if` clause changes the behaviour of the library for failed authorizations. We will be breaking also the existing integrations that properly deal with `null` values returned by the existing `getAuthorizationFlow()` getter

If we want to simply deprecate the usage of the existing method (created at compile time by the lombok library). I'd rather:
- leave the authorization flow object here, with protected visibility;
- move the lombok @Getter from this class to the Status instance variable only;
- create a new getter in both concrete classes (authorizing and failed) and tag the one on the failed case with @Deprecated 

With this approach and a good javadoc we can convey the right integration approach and at the same time keep the healthy/null-checking integration happy. Then, with a breaking release (i.e 3.0.0) we can remove the deprecated stuff. 

As an alternative we can release now already a 3.0.0 with simply the breaking change. 

WDYT?

Reply:
In other words, the change you're suggesting here is breaking at runtime as opposed to compile time. I'd rather compile-time break or not break (all the times) as of now